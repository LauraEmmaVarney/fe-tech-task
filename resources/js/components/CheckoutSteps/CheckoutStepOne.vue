<template>
    <base-card>
        <template v-slot:header>
            <div class="row align-items-center">
                <div class="col-auto">
                    <i class="fal fa-user-circle text-primary fa-2x"/>
                </div>
                <div class="col pl-0">
                    <h5 class="mb-0 font-weight-normal">Your Details</h5>
                </div>
            </div>
        </template>
        <template v-slot:body>
            <div 
                role="form"
                class="p-4"
            >
                <div class="form-group">
                    <label>Name</label>
                    <input 
                        v-model="form.name"
                        type="text" 
                        autocomplete="given-name" 
                        inputmode="text" 
                        placeholder="John Smith" 
                        autofocus
                        class="form-control" 
                    />
                </div>
                <div class="form-group">
                    <label>Email</label>
                    <div class="input-group">
                        <div class="input-group-prepend">
                            <span 
                                class="input-group-text" 
                                id="email-addon"
                            >
                                <i class="fal fa-envelope text-primary" />
                            </span>
                        </div>
                        <input 
                            v-model="form.email"
                            type="email" 
                            autocomplete="email" 
                            inputmode="email" 
                            placeholder="johnsmith@email.com" 
                            class="form-control" 
                        />
                    </div>
                </div>
            </div>
        </template>
        <template v-slot:footer>
            <div class="row align-items-center">
                <div class="col-md-6 col-lg-6 text-md-right">
                    <button 
                        type="submit" 
                        class="btn btn-primary btn-block" 
                        v-on:click="$emit('submitStepOne')"
                    >
                        Continue
                    </button>
                </div>
            </div>
        </template>
    </base-card>
</template>

<script>

import BaseCard from "../Utilities/BaseCard.vue";

    export default {
        name: "CheckoutStepOne",
        components: {
            BaseCard
        },
        props: {
            form: {
                type: Object,
                required: true
            }
        }
    }

</script>