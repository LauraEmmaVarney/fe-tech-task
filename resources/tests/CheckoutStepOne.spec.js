import { shallowMount } from '@vue/test-utils'
import CheckoutStepOne from '../js/components/CheckoutSteps/CheckoutStepOne.vue';

describe('CheckoutStepOne component tests', () => {
    const wrapper = shallowMount(CheckoutStepOne, {
        propsData: {
            form: {"name":"","email":"","card":""}
        }
    });

    it('mounts the component', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('checks props data', () => {
        expect(wrapper.props().form).toEqual({"name":"","email":"","card":""});
    });

    it('checks emitted events', () => {
        wrapper.find('button').trigger('click');
        expect(wrapper.emitted()).toHaveProperty('submitStepOne');
        expect(wrapper.emitted('submitStepOne')).toHaveLength(1);
    });

    it('sets and checks a value for the form', async () => {
        const nameInput = wrapper.find('input[type="text"]');
        await nameInput.setValue('John Smith');
        expect(nameInput.element.value).toBe('John Smith');
    })

    it('check against jest snapshot test', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});