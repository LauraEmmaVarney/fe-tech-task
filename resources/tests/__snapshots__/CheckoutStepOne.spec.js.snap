// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutStepOne component tests check against jest snapshot test 1`] = `
<base-card-stub>
  <div
    class="row align-items-center"
  >
    <div
      class="col-auto"
    >
      <i
        class="fal fa-user-circle text-primary fa-2x"
      />
    </div>
     
    <div
      class="col pl-0"
    >
      <h5
        class="mb-0 font-weight-normal"
      >
        Your Details
      </h5>
    </div>
  </div>
  <div
    class="p-4"
    role="form"
  >
    <div
      class="form-group"
    >
      <label>
        Name
      </label>
       
      <input
        autocomplete="given-name"
        autofocus="autofocus"
        class="form-control"
        inputmode="text"
        placeholder="John Smith"
        type="text"
      />
    </div>
     
    <div
      class="form-group"
    >
      <label>
        Email
      </label>
       
      <div
        class="input-group"
      >
        <div
          class="input-group-prepend"
        >
          <span
            class="input-group-text"
            id="email-addon"
          >
            <i
              class="fal fa-envelope text-primary"
            />
          </span>
        </div>
         
        <input
          autocomplete="email"
          class="form-control"
          inputmode="email"
          placeholder="johnsmith@email.com"
          type="email"
        />
      </div>
    </div>
  </div>
  <div
    class="row align-items-center"
  >
    <div
      class="col-md-6 col-lg-6 text-md-right"
    >
      <button
        class="btn btn-primary btn-block"
        type="submit"
      >
        
                    Continue
                
      </button>
    </div>
  </div>
</base-card-stub>
`;
